packages/iov-bns-governance/src/governor.spec.ts

```diff
@@ -0,0 +1,239 @@
+    pendingWithoutBnsd();
+    const options = await getConnectionAndIdentity();
+    const governor = new Governor(options);
+    expect(governor).toBeTruthy();
+
+    options.connection.disconnect();
+  });
+
+  describe("getElectorates", () => {
+    it("can get electorates", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      const electorates = await governor.getElectorates();
+      expect(electorates.length).toEqual(2);
+      expect(electorates[0].id).toEqual(1);
+      expect(electorates[1].id).toEqual(2);
+
+      options.connection.disconnect();
+    });
+
+    it("can get empty list of electorates", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity(bnsdUrl, HdPaths.iov(100));
+      const governor = new Governor(options);
+
+      const electorates = await governor.getElectorates();
+      expect(electorates.length).toEqual(0);
+
+      options.connection.disconnect();
+    });
+  });
+
+  describe("getElectionRules", () => {
+    it("throws for non-existent electorateId", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      const electorateId = 100;
+      return governor
+        .getElectionRules(electorateId)
+        .then(
+          () => fail("Expected promise to be rejected"),
+          err => expect(err.message).toMatch(/no election rule found/i),
+        )
+        .then(() => options.connection.disconnect());
+    });
+
+    it("returns the latest versions of election rules for an electorate", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      const electorateId = 1;
+      const electionRules = await governor.getElectionRules(electorateId);
+      expect(electionRules.length).toEqual(1);
+      expect(electionRules[0].id).toEqual(1);
+      expect(electionRules[0].version).toEqual(1);
+
+      options.connection.disconnect();
+    });
+  });
+
+  describe("getProposals", () => {
+    it("can get an empty list of proposals", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity(bnsdUrl, HdPaths.iov(100));
+      const governor = new Governor(options);
+
+      const proposals = await governor.getProposals();
+      expect(proposals.length).toEqual(0);
+
+      options.connection.disconnect();
+    });
+  });
+
+  describe("createProposalTx", () => {
+    it("throws an error for unsupported proposal types", async () => {
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      return governor
+        .createProposalTx({
+          type: ProposalType.AddCommitteeMember,
+          description: "Change something",
+          startTime: 1562161125,
```

Comment:
What about using `ReadonlyDate` instead of timestamps in the bns-governance interface? Seems to feel a bit less lowlevel.